ui/src/pages/backoffice/Loan/LoanSearch/SearchDateRange.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import { Card } from 'semantic-ui-react';
+import { DatePicker } from '@components';
+import { withState } from 'react-searchkit';
+import _isEmpty from 'lodash/isEmpty';
+import _flatten from 'lodash/flatten';
+
+class _SearchDateRange extends Component {
+  getCurrentDates() {
+    const { filters } = this.props.currentQueryState;
+    let fromDate = '';
+    let toDate = '';
+
+    filters.forEach(([name, value]) => {
+      if (name === 'loans_from_date') fromDate = value;
+      if (name === 'loans_to_date') toDate = value;
+    });
+    return [fromDate, toDate];
+  }
+
+  /** react-searchkit allows having the same filter multiple times with different values.
+   * For this range dates filters, we want each filter to appear only one time with one value
+   * (e.g. loan_start_date = `<date>`)
+   */
+  onDateChange = newFilter => {
+    const [name, value] = newFilter;
+    if (_isEmpty(value))
+      return this.props.updateQueryState({ filters: newFilter });
+
+    const oldFilters = this.props.currentQueryState.filters.filter(
+      filter => filter[0] === name
+    );
+
+    let filters = [newFilter];
+    if (!_isEmpty(oldFilters)) filters.unshift(_flatten(oldFilters));
```

Comment:
I am not sure I understand all this manipulation of values and different cases.
Here my reasoning:

case 1:
 * current: []
 * newFilter: ['loan_start_date', '2020-01-01']
 * call `this.props.updateQueryState({ filters: [ ['loan_start_date', '2020-01-01'] ] });`

case 2:
 * current: ['loan_start_date', '2020-01-01']
 * newFilter: ['loan_start_date', '2021-02-02']
 * call `this.props.updateQueryState({ filters: [ ['loan_start_date', '2020-01-01'], ['loan_start_date', '2021-02-02'] ] });`

In any case, we always pass the previous value to remove it, and eventually if a new one is set, add it.

```
onDateChange = newFilter => {
      const [name, value] = newFilter;
      const newFilters = this.props.currentQueryState.filters

      // if there is a new date, add it to the current filters so that the previous date is removed and the new one is added.
      // if the date is cleared, then just leave the previous date in the filters so that it will be remove
      const hasSelectedNewDate = !_isEmtpy(value);
      if (hasSelectedNewDate) {
          newFilters.push(newFilter)
      }

      return this.props.updateQueryState({ filters: newFilters });
}
```

Reply:
@ntarocco I see what you mean and I am sorry for the confusion it was very messy I cleaned it up according to your suggestions, thanks!